src/JuMP.jl

```diff
@@ -244,7 +244,8 @@ setPrintHook(m::Model, f) = (m.printhook = f)
 ###############################################################################
 # Variable class
 # Doesn't actually do much, just a pointer back to the model
-immutable Variable <: ReverseDiffSparse.Placeholder
+abstract AbstractVariable <: ReverseDiffSparse.Placeholder
```

Comment:
This is to allow `JuMPTypes` to be more general. It may also allow for other code savings in JuMPeR, but just focussed on linalg right now